### PR TITLE
create bucky resource

### PR DESCRIPTION
**Pull request by cjdcordeiro**

### CODE REVIEW

File: cimi-resources/src/com/sixsq/slipstream/ssclj/resources/storage_bucket.clj [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+   :rules [{:principal id
+            :type      "USER"
+            :right     "VIEW"}]})
+
+;;
+;; set the resource identifier to "storage-bucket/uuid(connector-href, instanceID)"
+;;
+(defmethod crud/new-identifier resource-name [json resource-name]
+  (let [credential-href (get (first (get json :credentials)) :href)
+        bucketName (get json :bucketName)
```
Comment (loomis) [inline cimi-resources/src/com/sixsq/slipstream/ssclj/resources/storage_bucket.clj]: Neither of these values of very idiomatic.  These would be better replaced by:
```
(-> json :credentials first :href)
```
and
```
(:bucketName json)
```
The functions `get` and `get-in` are avoided in most cases.

File: cimi-resources/test/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket_test.cljc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                :acl          valid-acl
+                :name         "short name"
+                :description  "short description",
+                :properties   {:a "one", :b "two"}
+                :bucketName   "aaa-bbb-111"
+                :usage        123.456
+                :connector    {:href "connector/0123-4567-8912"}
+                :credentials  [{:href  "connector/0123-4567-8912",
+                                :roles ["realm:cern", "realm:my-accounting-group"]
+                                :users ["long-user-id-1", "long-user-id-2"]}]
```
Comment (loomis) [inline cimi-resources/test/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket_test.cljc]: You can leave out the commas from vectors and lists.

File: cimi-resources/src/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket.cljc [hunk 3]
```diff
@@ -0,0 +1,26 @@
+(ns com.sixsq.slipstream.ssclj.resources.spec.storage-bucket
+  (:require
+    [clojure.spec.alpha :as s]
+    [com.sixsq.slipstream.ssclj.resources.spec.common :as cimi-common]
+    [com.sixsq.slipstream.ssclj.resources.spec.core :as cimi-core]
+    [com.sixsq.slipstream.ssclj.util.spec :as su]))
+
+(s/def ::href ::cimi-core/nonblank-string)  
+(s/def ::credentials (s/coll-of (s/keys :req-un [::href])))
```
[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply cimi-resources/src/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket.cljc]: I think this needs to be a list, so can it be a list of resource-links?

File: cimi-resources/src/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket.cljc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  (:require
+    [clojure.spec.alpha :as s]
+    [com.sixsq.slipstream.ssclj.resources.spec.common :as cimi-common]
+    [com.sixsq.slipstream.ssclj.resources.spec.core :as cimi-core]
+    [com.sixsq.slipstream.ssclj.util.spec :as su]))
+
+(s/def ::href ::cimi-core/nonblank-string)  
+(s/def ::credentials (s/coll-of (s/keys :req-un [::href]))) 
+(s/def ::bucketName ::cimi-core/nonblank-string)
+(s/def ::usage float?)
```
[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply cimi-resources/src/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket.cljc]: but then we need to save the usage in bytes...is that ok?

File: cimi-resources/src/com/sixsq/slipstream/ssclj/resources/storage_bucket.clj (hunk 1 above)
Comment (cjdcordeiro) [reply cimi-resources/src/com/sixsq/slipstream/ssclj/resources/storage_bucket.clj]: pushed

File: cimi-resources/test/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket_test.cljc (hunk 2 above)
Comment (cjdcordeiro) [reply cimi-resources/test/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket_test.cljc]: fixed

File: cimi-resources/src/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket.cljc (hunk 3 above)
[the remark this answers is not in the document]
Comment (loomis) [reply cimi-resources/src/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket.cljc]: `::cimi-common/resource-links` (with an 's') is a list of `::cimi-common/resource-link`.

File: cimi-resources/src/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket.cljc (hunk 4 above)
[the remark this answers is not in the document]
Comment (loomis) [reply cimi-resources/src/com/sixsq/slipstream/ssclj/resources/spec/storage_bucket.cljc]: I think that you can choose either KB or MB and still keep the value an integer/long.  The inaccuracy in doing this won't be noticeable.

**Comment by cjdcordeiro** [comment]
@loomis, should have addressed all comments

**Pull request merged**